### ISSUE TITLE
Removed event (on client) not receiving data from multiple remove service call

### ISSUE THREAD

**Issue by RickEyre** [feathersjs/feathers#843]
I'm using feathers, feathers-sequelize, feathers-primus, and feathers-client.

Remove multiple records via query:

```js
app.service('my-service').remove(null, {
  query: { $or: [{ id: 1 }, { id: 2}] }
});
```

Locally on the server what is expected happens.

```js
app.service('my-service').on('removed', data => console.log(data)); // <= [{ id: 1, /* props */ }, { id: 2, /* props */ }]
```

However, on my client.

```js
app.service('my-service').on('removed', data => console.log(data)); // <= null
```
I would think on the client I would receive the array of objects removed on the backend.

**Comment by daffl** [comment]
You would get a `removed` event for each entry but it definitely shouldn't be `null`. What is the return value of the `remove` call?

**Comment by RickEyre** [comment]
Thanks for your response @daffl 

The result of the remove call is the array of items removed.

> You would get a removed event for each entry but it definitely shouldn't be null

I only get one removed event for this call on the server. It receives the array of items. On the client it's null, but only one event.

**Comment by daffl** [comment]
What version of `feathers-sequelize`, `@featherjs/feathers` and other modules? Did you change the `raw` option to  `false`?

**Comment by RickEyre** [comment]
- feathers-sequelize@3.0.1
- @feathersjs/feathers@3.0.0
   - feathers-hooks-common@4.8.0
      - @feathersjs/feathers@3.1.3

I have set the `raw` option to `false` for some of my services, but this problem seems to not be isolated to just those ones.

**Comment by RickEyre** [comment]
@daffl any thoughts?

I can dive deeper on this if needed...

**Comment by RickEyre** [comment]
After some investigation I have found that it is happening [here](https://github.com/feathersjs/transport-commons/blob/master/lib/socket/utils.js#L49). Not sure why yet ...

**Comment by daffl** [comment]
Oooh... I bet it should be

```js
result = result[result.indexOf(data)];
```

**Comment by daffl** [comment]
Then we need some more information:

- What database and ORM are you using?
- Did you set Sequelize or Mongoose model `raw` setting to `false`?
- How does the `result` data look like?
- Are you setting `hook.dispatch`? If so, how does it look like?
- Are you setting result data with [channel.send](https://docs.feathersjs.com/api/channels.html#channelsenddata)? If so, how does it look like?

**Comment by RickEyre** [comment]
Okay, I've realized what I'm doing that's not working.

I've created a 'virtual' service that patches into other service events 'created', 'removed', 'updated', etc calls. And when I would receive these I would run some custom logic and then spit out the same service event on this 'vritual' service, but I would pass the array of data to the `emit` method.

I read [this piece of code](https://github.com/feathersjs/feathers/blob/master/lib/events.js#L16) that emits events one by one for an array; I've replicated this same logic and so [this code](https://github.com/feathersjs/transport-commons/blob/master/lib/socket/utils.js#L49) is now working.

Is this just a caveat for the built in service event names? Or are custom events also constrained by this?
